src/test/scala/org/broadinstitute/dsde/firecloud/service/TSVFileSupportSpec.scala

```diff
@@ -51,4 +55,30 @@ class TSVFileSupportSpec extends AnyFreeSpec with TSVFileSupport {
     }
   }
 
+  "setAttributesOnEntity" - {
```

Comment:
all of these tests need more detailed assertions. i just did bare minimum to have some reasonable confidence that things worked. i've also tested with a local orch and tsvs import correctly

i'll add more detailed tests before merging, of course

Reply:
Could you make sure to test with a local orch all the way through Rawls for the following case?
* workspace already has entities in it
* entities already have an attribute "foo", which could be a string or could be an array (pls test both)
* TSV upload attempts to write an array to attribute "foo"

I was looking at the entity-update code in Rawls as part of my other PRs, and wonder about some of the behavior: if a list already exists, does the TSV replace it or append to it? I would expect to replace it, but I think we may append? What about the case where a String exists in its place?

Reply:
Oh yikes. Yeah I think we append in the case of an existing list, and error in the case of an existing single value. I'll definitely double-check this.

Reply:
Just to continue the discussion of expected behavior:

I agree with you that it should replace an existing list. Maybe Orch should generate a `RemoveAttribute` in front of creating the array. I believe Rawls processes the `op`s in order, so this should work.

In the case of an existing single-value, I would expect it to throw an error because there's a type mismatch. If there's agreement on that, then I think we're probably all set here, but again I will double check the actual behavior later this morning.

Reply:
I've confirmed the following about the current state:

* uploading a TSV with an array does append to it rather than replace it.
* uploading a TSV with an array doesn't replace an existing single-value string, and throws this error: `AttributeName(default,sample_type) of OV-04-1336-TP is not a list`


It seems like sticking a `RemoveAttribute` op in front of the list add ops fixes both of these cases and allows for replacement. Do you think that's the appropriate behavior for both cases? I originally thought that it may be best to throw an error in the case of replacing a single value with a list, but I think I've changed my mind and replacement should be seamless and one step


```
            val removeOldListOp = Seq(Map(removeAttrOperation,nameEntry))
            removeOldListOp ++ addElements
```

Reply:
I agree on overwriting in either case, and the implementation of adding a `RemoveAttribute` first sounds good to me. I'd like to hear product's opinion on this too, but it all seems right to me.